smarts/core/utils/file.py

```diff
@@ -82,3 +85,26 @@ def file_md5_hash(file_path: str) -> str:
         hasher.update(f.read().encode())
 
     return str(hasher.hexdigest())
+
+
+@lru_cache(maxsize=1)
+def smarts_log_dir() -> str:
+    ## Following should work for linux and macos
+    # home_dir = os.path.expanduser("~")
+    # _, home_user = os.path.split(home_dir)
+    # temp_dir = os.path.join(tempfile.gettempdir(), "smarts")
+
+    # if not os.path.exists(temp_dir):
+    #     original_mask = os.umask(0)
+    #     try:
+    #         os.makedirs(temp_dir, mode=0o777)
+    #         os.chmod(temp_dir, mode=0o777)
+    #     finally:
+    #         os.umask(original_mask)
+
+    # return os.path.join(temp_dir, f"{home_user}")
```

Comment:
This above would work on linux and macos. The question would be windows.

Reply:
Can `tempfile` take care of the permission?

Reply:
If the intention is to have a one directory that is consistent across runs, then we shouldn't rely on /tmp/

Let's create a `.smarts` directory in the user's home directory and put sumo logs under there.

Reply:
> Can `tempfile` take care of the permission?
> 
> Sorry, I am not sure what's the problem here. I mean if `tempfile.mkdtemp` works well, why would we need to handle the permission?

There only consideration is that we are changing the file structure from the original to be completely flat. If that is already fine then there are no other issues.

Reply:
> If the intention is to have a one directory that is consistent across runs, then we shouldn't rely on /tmp/
> 
> Let's create a `.smarts` directory in the user's home directory and put sumo logs under there.

That can be done. The commented out solution uses a combination of the home directory name and `tempfile.gettempdir()`(which finds the same temp directory used by `tempfile.mkdtemp`.) The original reason for using temp over the home directory was to allow for automatic cleanup of the logs, which we could suspect can become quite large.